iothub/service/src/IotHubServiceClient.cs

```diff
@@ -125,6 +127,7 @@ public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHu
             _hostName = hostName;
             _httpClient = HttpClientFactory.Create(_hostName, options);
             _httpRequestMessageFactory = new HttpRequestMessageFactory(_credentialProvider.HttpsEndpoint, ApiVersion);
+            _options = options;
```

Comment:
It doesn't look like we need to save this options bundle in this class locally. All you need to do is pass it as an argument into the ```InitializeSubclients``` function below